taichi/transforms/offload.cpp

```diff
@@ -333,7 +333,8 @@ class IdentifyValuesUsedInOtherOffloads : public BasicStmtVisitor {
       return;
     auto top_level_ptr = SquashPtrOffset::run(stmt);
     // We don't support storing a pointer for now.
-    if (top_level_ptr->is<GlobalPtrStmt>())
+    if (top_level_ptr->is<GlobalPtrStmt>() || stmt->is<ExternalPtrStmt>() ||
```

Comment:
nit: can we wrap these checks into a helper function so that we can have a single place to change the logic in the future? ;)

Reply:
Completely solving the problem of identifying pointers will be done in #2546. As @xumingkuan said, only adding a helper function might not be good enough.
Meanwhile, we will be able to support pointers here (no need of these ad-hoc checks) once we improve our type system.